### PR TITLE
Calculator widget: Fixed buttonsize, resizing, num_pad_only option, decprecated items ...

### LINKED ISSUE #1436: GTK3: proposal for fixing button size in calculator widget
As my little hack for increasing the Ok and Cancel button for the calculator widget (https://github.com/LinuxCNC/linuxcnc/pull/1029) does not work any longer with Gtk3, here some possibilities for the new layout. What would you prefer?

We can have it close to the Gtk2 version (1):

or integrated (2):

or (I played a little bit around) a slighty bigger Ok button (3):

or (4):

or (5):

### PR DESCRIPTION
**Pull request by hansu**
- Fix size of OK and Cancel button for GTK3 (now integrated in button grid)
- Button field is now [resized with window ](https://user-images.githubusercontent.com/67957916/146643179-4b7345a6-6da8-4dc6-afec-91be4b1a7643.gif)

See https://github.com/LinuxCNC/linuxcnc/issues/1436

closes #1436

@c-morley 

The last commit includes this layout:

For those who are more adventurous, they can revert the last commit and try this version:

However the layout can simply be modified by clicking with glade-3.

### CODE REVIEW

**Comment by hansu** [comment]
@c-morley Can you please check if the buttons are correct when calling  `restart_dialog(self)` from gscreen.py?
It seems that you only have a Cancel button there, and maybe there is an Ok button that is unwanted?

**Comment by c-morley** [comment]
Sorry I didn't notice this. In Gscreen I see to sets of Ok and Cancel buttons in the tool change dialog

**Comment by c-morley** [comment]
Looks like gladevcp is still fairly broken - the hal_action/hal_filechooser errors.

**Comment by c-morley** [comment]
gscreen_tool_dialog

**Comment by hansu** [comment]
Did you check out the latest version? If I do this, it is displayed correctly:

grafik

And what do you mean with 
>Looks like gladevcp is still fairly broken - the hal_action/hal_filechooser errors. 

?

**Comment by c-morley** [comment]
I pulled this branch.
last commit c821586

the gladevcp commit is unrelated to your changes. I tried out gscreen and couldn't load a program because (at least seemingly )of errors in gladevcp's file chooser

**Comment by hansu** [comment]
Strange. It seems that you don't have the changes from https://github.com/LinuxCNC/linuxcnc/pull/1454/commits/116cbd1d10176eaa40784fb410c6518302ae5f97


>the gladevcp commit is unrelated to your changes. I tried out gscreen and couldn't load a program because (at least seemingly) of errors in gladevcp's file chooser

Ah ok

**Comment by c-morley** [comment]
I'm not a git hub master ut when I clicked on the missing commit, I see on the top a commant:

This commit does not belong to any branch on this repository, and may belong to a fork outside of the repository.

**Comment by hansu** [comment]
Yeah Github changes the link when pasting it here, that is strange.
Do you added the remote `git@github.com:hansu/linuxcnc.git `and checked out the branch `calculatorwidget-gtk3`?

**Comment by c-morley** [comment]
Yes, and just repulled -it's up to date.

**Comment by hansu** [comment]
Strange. Maybe try `make clean` before? Otherwise, I have no idea. Maybe someone else could try this.

**Comment by hansu** [comment]
> Looks like gladevcp is still fairly broken - the hal_action/hal_filechooser errors.

is now fixed

**Comment by c-morley** [comment]
The Dialog looks right now and the file chooser is fixed. thanks

**Comment by c-morley** [comment]
Looking at the run from line dialog - ya it looks busy. It's been so long that I don't remember how it was exactly intended to work.
But having enter and ok seems wrong. having units conversion button seems wrong.
If I was to do it today I think I would build a specific dialog for number entry rather then use a calculator widget.

**Comment by hansu** [comment]
>having units conversion button seems wrong.

Yes that's my fault and I already noticed that. I added the conversion buttons some time ago and didn't know about the `num_pad_only()` - fixed now.

>But having enter and ok seems wrong. 

Exactly. We can either have the functionally of that button like it is in Gmoccapy or we remove the enter button.
Im Gmoccapy, the enter button works as a "apply-button" - so jump to the line but let the window open. What would you prefer?

> If I was to do it today I think I would build a specific dialog for number entry rather then use a calculator widget.

Yeah that might would have been easier. But it also has some advantages to have only one.

Do you think we can give this dialog a border and titlebar etc. again? 
You made it 9 years ago to force a bigger size, but i think this isn't needed any longer.
grafik


**Still some room for optimizations, so please don't merge until I give the ok**

**Comment by c-morley** [comment]
One reason to remove the title bar is that in fullscreen, if there is a title bar some (all?) window managers will show the top and bottom panel. Though it's more annoying then a problem. the fact it's difficult to move the dialog is probably more a problem,

**Comment by hansu** [comment]
>One reason to remove the title bar is that in fullscreen, if there is a title bar some (all?) window managers will show the top and bottom panel. 

Ok yeah, that resolves this problem in some cases. Unfortunately not in Gmoccapy.

The restart entry looks now so:

image

In Gmoccapy, the Enter button is no the Jump to button.
In Gscreen, the Enter button is equal to the Ok button. The Jump to button is new and does the same as in Gmoccapy.
I didn't touch the behaviour when pressing Enter in the entry field. In Gmoccapy it jumps to the line (=Jump to button) and in Gscreen it jumps to the line and closes the window (=Ok button). Chris you decide if wee keep this ;-)

**Comment by hansu** [comment]
From my side this PR is ready to merge if you agree with the behaviour in https://github.com/LinuxCNC/linuxcnc/pull/1454#issuecomment-1004101382 :slightly_smiling_face:

File: src/po/Submakefile [hunk 1]
```diff
@@ -55,13 +55,14 @@ TCLSRCS := \
 	../tcl/tooledit.tcl
 
 TOOLI18NSRCS := \
-    ../lib/python/gladevcp/iconview.py \
-    ../lib/python/gladevcp/hal_mdihistory.py \
+	../lib/python/gladevcp/iconview.py \
```
Comment (c-morley) [inline src/po/Submakefile:58]: Is this a tab vrs space change?
I understand adding calculator.glade but don't understand the other changes.

Comment (hansu) [reply src/po/Submakefile:58]: >Is this a tab vrs space change?

Exactly! Only space changes except adding calculator.glade. Which other changes do you not understand?
I just recognized that I accidentally commited some files (gmoccapy.pot, linuxcnc.pot, sim.tbl) - maybe these ones?
I already corrected this.

**Comment by c-morley** [comment]
Thabk you!

**Pull request merged**